src/hooks/useQuery.ts

```diff
@@ -0,0 +1,35 @@
+import { QueryKey, UseQueryOptions } from "react-query";
+import { useQuery as useRQQuery } from "react-query";
+
+import useQueryClient from "./useQueryClient";
+
+import { GraphQLErrorType } from "src/types/GraphQLErrorType";
+
+interface useQueryProps<TQuery, TVariables> {
+    queryKey: QueryKey;
+    query: string;
+    variables?: TVariables;
+    options?: UseQueryOptions<TQuery, GraphQLErrorType>;
+}
+
+const useQuery = <TQuery, TVariables>({
+    queryKey,
+    query,
+    options,
+    variables,
+}: useQueryProps<TQuery, TVariables>) => {
+    const queryClient = useQueryClient();
+
+    return useRQQuery<TQuery, GraphQLErrorType>(
+        queryKey,
+        async () => {
+            const result = await queryClient.request<TQuery, TVariables>(query, variables);
+            return result;
+        },
+        {
+            ...options,

```

Comment:
Why use rest parameter here?

Reply:
Same as useMutation, for me is that using Rest Parameter is the most convenient way to pass all options properties to useRQQuery.